app/services/contributors_parser.rb

```diff
@@ -0,0 +1,23 @@
+class ContributorsParser
+  attr_reader :data, :legislator_id
+
+  def initialize(data, legislator_id)
+    @data = data
+    @legislator_id = legislator_id
+  end
+
+  def parse
+    contributors = data.map do |contributor_data|
+      Contributor.where(legislator_id: legislator_id, sunlight_id: contributor_data[:id]).first_or_create.tap do |contributor|
```

Comment:
Looks like you might not be saving new contributors (or capturing all their data).